pipelines/smartseq2_single_sample/SS2SingleSample.wdl

```diff
@@ -0,0 +1,136 @@
+##### Naming Rules #####
+## ** Files should be named in CamelCase starting with a capital
+## ** Tasks should be named in CamelCase starting with a capital
+## ** Alias' should be named in CamelCase starting with a capital
+## ** Variable declarations should be lower_snake_case
+
+##### New Line Rules #####
+## ** New line before and after `calls`
+## ** New Line between task definitions
+## ** New lines between blocks in tasks
+## ** New line variable declarations that you want to group
+## ** New line between imports and workflow definition
+## ** No new lines at the start or end of any workflow/call/task definition block
+## ** No need for having two new lines anywhere
+## ** No new lines at the start of any blocks
+
+##### Spacing Rules #####
+## ** Space before `{` characters
+## ** Space after `,` characters
+## ** Space before and after operators e.g. `+`, `=`
+## ** Space after `:` character in runtime block
+## ** Space after `#` when creating a comment (or `##` if we decide thats our comment definer)
+
+##### Tab vs Space for Indents ######
+## ** Always use spaces? I think whats most frequently used is whenever you want to indent you use 2 spaces but we should decide what works best for us.
+
+##### Runtime Attributes ######
+## ** This will change quite a bit i believe when the new spec of WDL comes out (WDL SPEC 3) but we should define all values we care about
+## ** even if they are the default values.  e.g. write `cpu: 1` instead of leaving it out because its the default.
```

Comment:
This is subtly different from leaving it unsaid.

Leaving it unsaid allows a runner to override it whereas I believe specifying it hard-codes it into the WDL.

For things like `failOnStderr` that *should* be baked into the workflow whereas for things like `cpu` it should be up to whoever's running it.

To allow overrides you can have runtime fed by inputs with defaults, eg
```
task foo {
  Int cpus
  runtime {
    cpu: cpus
  }
}
```

Reply:
But if its a single threaded tool why would you want to allow the runner to override it?  I think theres definitely cases where you want it to be overridable but my original comment was more so talking about not relying on defaults set by Cromwell, i.e. if you don't set a `cpu` value for the PAPI backend you always get a value of `1`.  I don't even know the defaults for local or any other backend which is kind of my point because if we had set a value in our task I woudln't have to worry about that either.

Reply:
I like baking in CPUs and Memory because very often this just means that you're coding in the maximum allotment for the machine type that you require. 

i.e. if you need 30gb of ram and you have a multi-processed tool, you may as well request the 8 cores you get with that. 

Interested to hear counter-examples where this is not productive. I do have some WDLs where things are set variably (e.g. alignment, where ram requirements are a function of genome index size).